gnomad_qc/v3/subset/subset.py

```diff
@@ -0,0 +1,243 @@
+            "Description": "Sum of PL[0] values; used to approximate the QUAL score"
+        },
+        "MQ_DP": {
+            "Description": "Depth over variant samples for better MQ calculation"
+        },
+        "VarDP": {"Description": "(informative) depth over variant genotypes"},
+        "SB": {
+            "Description": "Per-sample component statistics which comprise the Fisher's Exact Test to detect strand bias."
+        },
+        "MQ": {
+            "Description": "Root mean square of the mapping quality of reads across all samples"
+        },
+        "QD": {
+            "Description": "Variant call confidence normalized by depth of sample reads supporting a variant"
+        },
+        "FS": {
+            "Description": "Phred-scaled p-value of Fisher's exact test for strand bias"
+        },
+    },
+}
+
+
+def format_info_for_vcf(ht) -> hl.Table:
+    """Formats gnomAD MT fields with types that are not condusive to vcf export
+    :param ht: Table to reformat
+    :return: Reformatted Table
+    """
+    logger.info("Formatting the info hail table for VCF export")
+    for f, ft in ht.info.dtype.items():
+        if ft == hl.dtype("int64"):
+            ht = ht.annotate(
+                info=ht.info.annotate(**{f: hl.int32(hl.min(2 ** 31 - 1, ht.info[f]))})
+            )
+        elif ft == hl.dtype("float64"):
+            ht = ht.annotate(
+                info=ht.info.annotate(
+                    **{f: hl.float32(hl.min(2 ** 31 - 1, ht.info[f]))}
+                )
+            )
+        elif ft == hl.dtype("array<int64>"):
+            ht = ht.annotate(
+                info=ht.info.annotate(
+                    **{f: ht.info[f].map(lambda x: hl.int32(hl.min(2 ** 31 - 1, x)))}
+                )
+            )
+        elif ft == hl.dtype("array<float64>"):
+            ht = ht.annotate(
+                info=ht.info.annotate(
+                    **{f: ht.info[f].map(lambda x: hl.float32(hl.min(2 ** 31 - 1, x)))}
+                )
+            )
+    return ht
+
+
+def compute_partitions(mt, entry_size=3.5, partition_size=128000000) -> int:
+    """Computes a very rough estimate for the optimal number of partitions in a MT
+     using a the hail recommended partition size of 128MB and the rough estimate 
+     of 3.5B per entry in the gnomAD sparse MT.
+
+    :param mt: MatrixTable that will be resized
+    :param entry_size: Average size in bytes that a single entry requires , defaults to 3.5
+    :param partition_size: Amount of data per partition. Hail recommends 128MB, defaults to 128000000
+    :return: number of partitions for resizing MT  
+    """
+    rows, columns = mt.count()
+    mt_disk_est = rows * columns * entry_size
+    n_partitions = (
+        int(mt_disk_est / partition_size) if (mt_disk_est / partition_size) > 20 else 20
+    )
+    return n_partitions
+
+
+def main(args):
+    hl.init(
+        log="/subset.log", tmp_dir="hdfs:///subset.tmp/", default_reference="GRCh38"
```

Comment:
I double-checked this with Tim: `/tmp` (the hail default) is the same as `hdfs:///tmp`. Happy to keep the `tmp_dir` arg here if you want but just thought I'd clarified this as I think that there is some confusion about this (and the fact that using the default uses HDFS / SSDs if present)